forem/forem
# Prepare the database for the configurable navigation items

## What type of PR is this? (check all applicable)


## Description

This PR tackles the following subtasks related to providing admins with a configurable navbar:

1. Adding a field called order that allows an admin to decide on the order he'd like the navigation items to follow ( A future iteration can possibly be a drag and drop)
2. Renames a column more applicably 
3. Creates a data script to seed some default navigation items for all forems on creation
4. Creates a temporary data script that will allow DEV and existing forems like thismma and forem.dev to seed their navigational items.

## Related Tickets & Documents

Follow up PR of on https://github.com/forem/forem/pull/9444
https://github.com/forem/forem/issues/4661
https://github.com/forem/forem/issues/5475

## QA Instructions, Screenshots, Recordings

1. Run rake db:migrate 
2. Update the data scripts using `rails data_updates:run`. You should now see 5 items in the database using `NavigationLink.all.length'
3. Run the rake task using `rake temporary:navigation_links:update_links`. You should now see 13 items in the database using `NavigationLink.all.length'
4. You should also be able to add, edit and delete any link from http://localhost:3000/admin/navigation_links

Also triple check urls and names compared to the current site.

## Added tests?


## Added to documentation?


## [optional] Are there any post deployment tasks we need to perform?

1. Need to ensure that data script runs and check blazer if the NavigationLinks table is populated with 5 items
2. Run the rake task `rake temporary:navigation_links:update_links` and check that the NavigationLinks table is populated with 13 items

## [optional] What gif best describes this PR or how it makes you feel?

![alt_text](https://media.giphy.com/media/ggPs9xLsrkE2SsKlEl/giphy.gif)
